src/modules/GroupChannelList/context/GroupChannelListProvider.tsx

```diff
@@ -125,45 +122,32 @@ export const GroupChannelListProvider = (props: GroupChannelListProviderProps) =
     isTypingIndicatorEnabledOnChannelList = false,
     isMessageReceiptStatusEnabledOnChannelList = false,
   } = config;
-  const sdk = sdkStore?.sdk;
+
+  const sdk = sdkStore.sdk;
+  const isConnected = useOnlineStatus(sdk, config.logger);
+  const scheduler = useMarkAsDeliveredScheduler({ isConnected }, config);
 
   const channelListStore = useGroupChannelList(
     sdk,
-    !channelListQuery ? {} : {
-      collectionCreator: () => {
-        const filter = new GroupChannelFilter();
-        filter.includeEmpty = channelListQuery.includeEmpty ?? false;
-        filter.includeFrozen = channelListQuery.includeFrozen ?? false;
-        filter.setUserIdsFilter(
-          channelListQuery.userIdsExactFilter ?? channelListQuery.userIdsIncludeFilter ?? [],
-          channelListQuery.userIdsExactFilter ? false : true,
-          channelListQuery.userIdsIncludeFilterQueryType ?? QueryType.AND,
-        );
-        filter.nicknameContainsFilter = channelListQuery.nicknameContainsFilter ?? '';
-        filter.channelNameContainsFilter = channelListQuery.channelNameContainsFilter ?? '';
-        filter.customTypesFilter = channelListQuery.customTypesFilter ?? [];
-        filter.customTypeStartsWithFilter = channelListQuery.customTypeStartsWithFilter ?? '';
-        filter.channelUrlsFilter = channelListQuery.channelUrlsFilter ?? [];
-        filter.superChannelFilter = channelListQuery.superChannelFilter ?? SuperChannelFilter.ALL;
-        filter.publicChannelFilter = channelListQuery.publicChannelFilter ?? PublicChannelFilter.ALL;
-        filter.myMemberStateFilter = channelListQuery.memberStateFilter ?? MyMemberStateFilter.ALL;
-        filter.hiddenChannelFilter = channelListQuery.hiddenChannelFilter ?? HiddenChannelFilter.ALL;
-        filter.unreadChannelFilter = channelListQuery.unreadChannelFilter ?? UnreadChannelFilter.ALL;
-        return sdk.groupChannel.createGroupChannelCollection({
-          filter,
-          limit: channelListQuery.limit ?? 20,
-          order: channelListQuery.order ?? GroupChannelListOrder.LATEST_LAST_MESSAGE,
+    {
+      markAsDelivered: (channels) => channels.forEach(scheduler.push),
+      collectionCreator: getCollectionCreator(sdk, channelListQuery),
+      onChannelsDeleted: (channelUrls) => {
+        channelUrls.forEach((url) => {
+          if (url === selectedChannelUrl) onChannelSelect(null);
         });
       },
     },
   );
-  const {
-    refreshing,
-    initialized,
-    groupChannels,
-    refresh,
-    loadMore,
-  } = channelListStore;
+
+  const { refreshing, initialized, groupChannels, refresh, loadMore } = channelListStore;
+
+  // Auto select channel
+  useEffect(() => {
+    if (!disableAutoSelect && stores.sdkStore.initialized && initialized) {
+      if (!selectedChannelUrl) onChannelSelect(groupChannels[0] ?? null);
```

Comment:
minor: not sure which is better but we can join `!selectedChannelUrl` with above
```
if (!disableAutoSelect && stores.sdkStore.initialized && initialized && !selectedChannelUrl) {
```

Reply:
According to the logic, I separated it with a separate if statement.